src/pint/toa.py

```diff
@@ -2037,11 +2062,10 @@ def write_TOA_file(
 
         # If pulse numbers were added to flags, remove them again
         if pnChange:
-            for flags in self.table["flags"]:
-                try:
-                    del flags["pn"]
-                except KeyError:
-                    pass
+            self.delete_flag("pn")
+        # Same for delta pulse numbers
+        if dpnChange:
```

Comment:
Without a `try: ... finally:` the way this function works is highly prone to leaving TOAs objects in broken states when exceptions occur, including KeyboardInterrupt. Can all these changes be made on  a fresh copy of the TOAs object, which is written out and then discarded?

Reply:
This is particularly problematic for `-padd` - what happens if you find yourself with a TOAs object with both a `delta_pulse_numbers` column and some `-padd` flags - are the `-padd` redundant or are they to be added into the `delta_pulse_numbers`?

Reply:
My main concern with the second was that when the code was temporarily modifying the initial object, a control-C could leave it with both. Clock corrections are in the same boat. But if the object is not temporarily modified, we should be fine.